### ISSUE TITLE
Site Search spelling suggestions are un available in the response

### ISSUE THREAD

**Issue by JasonStoltz** [elastic/search-ui#436]
Original question:

> Hi there!
> My use case needs spelling suggestions from a site search engine. I can alter the query options:

```
connector = new SiteSearchAPIConnector({
    engineKey: xxxx,
    documentType: xxxx    
    beforeSearchCall: (queryOptions, next)=> {
      const withSpelling = {...queryOptions, spelling: 'strict'}
      return next(withSpelling)
    }
})
```

> And I can see the spelling suggestion coming back in the info field of the response but I don't see any way to surface this in search-ui. I'm guessing this isn't supported right now but if I've missed something obvious, please let me know :)

We unfortunately have no work-around for this. I think we could likely easily add a "didYouMean" field to state, though, which the site search connector could populate with this spelling suggestion.

Because it is a new field, it would be a non-breaking change. This feature is not supported by app search but could easily be added in the future if that feature becomes available.

We would also probably want to add search query configuration for this as well. I am imagining a configuration like:

```
config = {
  searchQuery: {
    didYouMean: { // Using an object to leave room for additional config in the future
      enabled: true
    }
  }
}
```

**Comment by Essk** [comment]
I'm wondering if it would be easier to build the request if the config directly mirrors the site-search options: 'strict', 'always', 'retry'?

```
config = {
  searchQuery: {
    didYouMean: { // Using an object to leave room for additional config in the future
      enabled: true
      spelling: ['strict', 'always', 'retry']
    }
  }
}
```

**Comment by JasonStoltz** [comment]
@Essk Yes, something like that makes sense to me.

Maybe in that case it would be better to do something like:

```
config = {
  searchQuery: {
    spelling: {
      type: 'strict' // 'strict', 'always', 'retry'
    }
  }
}
```

**Comment by davidsauntson** [comment]
Can I re-open this issue please? As I would like to finish this off in a fork of @Essk fork.  Thanks!

**Comment by JasonStoltz** [comment]
Yep, thanks!

**Comment by JasonStoltz** [comment]
Closed by #517
